airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java

```diff
@@ -136,7 +131,8 @@ private static void setJsonField(ResultSet r, int i, ObjectNode o) throws SQLExc
       case BIGINT -> o.put(columnName, nullIfInvalid(() -> r.getLong(i)));
       case FLOAT, DOUBLE -> o.put(columnName, nullIfInvalid(() -> r.getDouble(i), Double::isFinite));
       case REAL -> o.put(columnName, nullIfInvalid(() -> r.getFloat(i), Float::isFinite));
-      case NUMERIC, DECIMAL -> o.put(columnName, nullIfInvalid(() -> r.getBigDecimal(i)));
+      case NUMERIC -> o.put(columnName, nullIfInvalid(() -> r.getInt(i)));
+      case DECIMAL -> o.put(columnName, nullIfInvalid(() -> r.getBigDecimal(i)));
```

Comment:
@cgardens this way I don't need to change the `JdbcStandardTest`. Can you validate this change? I can't measure the impact of this change.